src/core/js/a11y/scroll.js

```diff
@@ -0,0 +1,304 @@
+                var scrollingParent = this._getScrollingParent(event, $target);
+                if (scrollingParent.filter(this._scrollDisabledElements).length === 0) {
+                    this.$window.scroll();
+                    return;
+                }
+            }
+            event.preventDefault();
+            return false;
+        },
+
+        /**
+         * Return the parent which will be scrolling from the current scroll event.
+         *
+         * @param {Object} event jQuery event object.
+         * @param {Object} $target jQuery element object.
+         */
+        _getScrollingParent: function(event, $target) {
+            var isTouchEvent = event.type == 'touchmove';
+            var hasTouchStartEvent = (this._touchStartEventObject && this._touchStartEventObject.originalEvent);
+            if (isTouchEvent && !hasTouchStartEvent) {
+                return $target;
+            }
+            var directionY = this._getScrollDirection(event);
+            if (directionY === 'none') {
+                return this.$body;
+            }
+            var parents = $target.parents();
+            for (var i = 0, l = parents.length; i < l; i++) {
+                var $parent = $(parents[i]);
+                if ($parent.is('body')) {
+                    return this.$body;
+                }
+                if (!this._isScrollable($parent)) {
+                    continue;
+                }
+                if (!this._isScrolling($parent, directionY)) {
+                    continue;
+                }
+                return $parent;
+            }
+            return this.$body;
+        },
+
+        /**
+         * Returns true if the specified target is scrollable.
+         *
+         * @param {Object} $target jQuery element object.
+         * @returns {boolean}
+         */
+        _isScrollable: function($target) {
+            var scrollType = $target.css('overflow-y');
+            if (scrollType !== 'auto' && scrollType !== 'scroll') {
+                return false;
+            }
+            var pointerEvents = $target.css('pointer-events');
+            if (pointerEvents === 'none') {
+                return false;
+            }
+            return true;
+        },
+
+        /**
+         * Returns true if the specified target is the scrolling target.
+         *
+         * @param {Object} $target jQuery element object.
+         * @param {string} directionY 'none' | 'up' | 'down'
+         *
+         * @returns {boolean}
+         */
+        _isScrolling: function($target, directionY) {
+            var scrollTop = Math.ceil($target.scrollTop());
+            var innerHeight = $target.outerHeight();
+            var scrollHeight = $target[0].scrollHeight;
+            var hasScrollingSpace = false;
+            switch (directionY) {
+                case 'down':
+                    hasScrollingSpace = scrollTop + innerHeight < scrollHeight;
+                    if (hasScrollingSpace) {
```

Comment:
if it's deliberate that it should fall through to the 'up' case (if `hasScrollingSpace` is `false`), could you mark it with a /* falls through */ to stop linters from flagging this - and maybe to provide a short explanation why

Reply:
done